### ISSUE TITLE
Doesn't insert node_modules/.bin for all parents

### ISSUE THREAD

**Issue by Arcanemagus** [timoxley/npm-path#2]
According to the description of this package it should be adding `node_modules/.bin` to the path for _all_ parent directories of the current directory, but it seems to only be doing that for `node_modules` directories.

Am I misunderstanding the purpose of this package, or is this a bug?

```
PS C:\Temp\npm-path-test> ls -recurse


    Directory: C:\Temp\npm-path-test


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----       2015-12-04     14:23                lib
d-----       2015-12-04     14:17                node_modules


PS C:\Temp\npm-path-test> npm-path
C:\Program Files\nodejs\node-gyp-bin;C:\Temp\npm-path-test\node_modules\.bin;C:\Program Files\nodejs;%PATH%
PS C:\Temp\npm-path-test> cd lib
PS C:\Temp\npm-path-test\lib> npm-path
C:\Program Files\nodejs\node-gyp-bin;C:\Temp\npm-path-test\lib\node_modules\.bin;C:\Program Files\nodejs;%PATH%
PS C:\Temp\npm-path-test\lib> cd ..\node_modules\
PS C:\Temp\npm-path-test\node_modules> npm-path
C:\Program Files\nodejs\node-gyp-bin;C:\Temp\npm-path-test\node_modules\node_modules\.bin;C:\Temp\npm-path-test\node_modules\.bin;C:\Program Files\nodejs;%PATH%
```

_Note: For brevity I replaced the normal path with `%PATH%`._

For a "real world" example, and what lead me here, this is what I believe _should_ be working, but isn't due to this:
![image](https://cloud.githubusercontent.com/assets/427137/11603429/851f5ae8-9a96-11e5-92bf-1b7ec9a610c8.png)
At first we get the package local `eslint` installation from `npm-which` (which uses this for PATH generation), but on moving into the `lib` folder we get the globally installed version instead of the package local version.

**Comment by timoxley** [comment]
@Arcanemagus it works as expected, follows the same algorithm as npm.

**Comment by timoxley** [comment]
Hm, perhaps it should pick up the local bin. I'll investigate later this week.

**Comment by timoxley** [comment]
@Arcanemagus no time! If you submit a PR for this, I'll merge it.

**Comment by Arcanemagus** [comment]
I believe we went with a different method within `linter-eslint`, however if you want I can take a look at some point. Rather busy myself supporting the Atom linters though so it will be a while before I can get to it.

**Comment by timoxley** [comment]
@Arcanemagus yep, would appreciate if you took a look at it. Curious, what was the alternative method you took?

**Comment by Arcanemagus** [comment]
It [looks like](https://github.com/AtomLinter/linter-eslint/blob/v6.0.0/src/worker-helpers.js#L58) accessing an internal method of the `module` object was used, @steelbrain could tell you more though.

**Comment by steelbrain** [comment]
I think we have a misunderstanding here. That `Module._initPaths()` makes sure that the latest changes to `NODE_PATH` are acknowledged by node :)

**Comment by timoxley** [comment]
@Arcanemagus @steelbrain note this is now fixed in npm-path@1.1.0 Thanks for your patience!
